reana_server/scheduler.py

```diff
@@ -138,9 +153,12 @@ def on_message(self, workflow_submission, message):
         """On new workflow_submission event handler."""
         message.ack()
         workflow_submission = json.loads(workflow_submission)
-        if reana_ready():
+        workflow_min_job_memory = workflow_submission.pop("min_job_memory", 0)
+        reana_memory_available = check_memory_availability(workflow_min_job_memory)
+        if reana_ready() and reana_memory_available:
```

Comment:
Is it OK to move `reana_memory_available`/`check_memory_availability` to `reana_ready()`?

Reply:
I haven't moved it there since it requires an argument

Reply:
We can pass the arg to `reana_ready` I guess. Then we can create a [partial](https://docs.python.org/3/library/functools.html#functools.partial) function including the arg and include it in the array of function conditions.